modules/combined/test/tests/beam_eigenstrain_transfer/master.i

```diff
@@ -0,0 +1,224 @@
+# Test for multi app vector postprocessor to aux variable transfer
+
+# Master App contains 2 beams, each fixed at one end and oriented
+# in the z direction. Each subApp contains a 2D model with the same
+# height as the length of the beam. A time varying temperature profile
+# is assigned to each 2D model and the resulting yy strain along the
+# height of the 2D model is transferred to the beam model using the
+# multi app transfer. The resulting displacement of the beam should be
+# same as that of the 2D models.
+
+[Mesh]
+  type = FileMesh
+  file = 2_beams.e
+[]
+
+[Variables]
+  [./disp_x]
+  [../]
+  [./disp_y]
+  [../]
+  [./disp_z]
+  [../]
+  [./rot_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./rot_y]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./rot_z]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[BCs]
+  [./fixx1]
+    type = DirichletBC
+    variable = disp_x
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixy1]
+    type = DirichletBC
+    variable = disp_y
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixz1]
+    type = DirichletBC
+    variable = disp_z
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixr1]
+    type = DirichletBC
+    variable = rot_x
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixr2]
+    type = DirichletBC
+    variable = rot_y
+    boundary = 1
+    value = 0.0
+  [../]
+  [./fixr3]
+    type = DirichletBC
+    variable = rot_z
+    boundary = 1
+    value = 0.0
+  [../]
+[]
+
+[Kernels]
+  [./solid_disp_x]
+    type = StressDivergenceBeam
+    block = '1'
+    displacements = 'disp_x disp_y disp_z'
+    rotations = 'rot_x rot_y rot_z'
+    component = 0
+    variable = disp_x
+  [../]
+  [./solid_disp_y]
+    type = StressDivergenceBeam
+    block = '1'
+    displacements = 'disp_x disp_y disp_z'
+    rotations = 'rot_x rot_y rot_z'
+    component = 1
+    variable = disp_y
+  [../]
+  [./solid_disp_z]
+    type = StressDivergenceBeam
+    block = '1'
+    displacements = 'disp_x disp_y disp_z'
+    rotations = 'rot_x rot_y rot_z'
+    component = 2
+    variable = disp_z
+  [../]
+  [./solid_rot_x]
+    type = StressDivergenceBeam
+    block = '1'
+    displacements = 'disp_x disp_y disp_z'
+    rotations = 'rot_x rot_y rot_z'
+    component = 3
+    variable = rot_x
+  [../]
+  [./solid_rot_y]
+    type = StressDivergenceBeam
+    block = '1'
+    displacements = 'disp_x disp_y disp_z'
+    rotations = 'rot_x rot_y rot_z'
+    component = 4
+    variable = rot_y
+  [../]
+  [./solid_rot_z]
+    type = StressDivergenceBeam
+    block = '1'
+    displacements = 'disp_x disp_y disp_z'
+    rotations = 'rot_x rot_y rot_z'
+    component = 5
+    variable = rot_z
+  [../]
+[]
+
+[Materials]
+  [./elasticity]
+    type = ComputeElasticityBeam
+    youngs_modulus = 2.60072400269
+    poissons_ratio = -0.9998699638
```

Comment:
The constants for youngs_modulus and poissons_ratio are kind of odd. Where do they come from?

Reply:
The values of E, A and I and length of beam are based on the timoshenko beam example from Prathap and Bhasyam (1982) and the example is in beam/static/timoshenko_small_strain_y.i

In beam/static/euler_small_strain_y.i, I converted the timoshenko beam to a Euler-Bernoulli beam by keeping the geometry of the beam constant but by varying the shear modulus. Basically if alpha (= kAGL^2/EI) is a large number, the beam would behave like a Euler-Bernoulli beam irrespective of whether the beam is slender or thick. For alpha = 2e6, the shear modulus has to be 1e4 which results in a poisson's ratio of -0.9998. This is within the allowable range for poisson's ratio.

Reply:
If you could put in a short comment to that effect in this input file, that would be helpful. Since this is a test of the eigenstrain transfer and not of the behavior of the beam, it also might be worth considering using a Timoshenko beam in this example.